File: web-sources/S8-fwks-palm/boxes/b2/B2ColumnsStruct.js


import { S8 } from "/S8-api/S8Context.js";
import { S8Object } from "/S8-api/S8Object.js";

import { B2Column } from "./B2Column.js";

import { S8WebFront } from "/S8-pkgs-ui-carbide/S8WebFront.js";


S8.page.CSS_import('/S8-fwks-palm/boxes/b2/B2Box.css');


/**
 * 
 */
export class B2ColumnsStruct extends S8Object {

    /** @type {HTMLDivElement} */
    wrapperNode;


    /** */
    constructor() {
        super();
        this.wrapperNode = document.createElement("div");
        this.wrapperNode.classList.add("b2struct-columns");
    }


    /** @returns */
    getEnvelope() {
        return this.wrapperNode;
    }



    /** @returns{S8Object[]} */
    getComponents() {
        return this.columns;
    }



    /** @param {B2Column[]} columns */
    S8_set_columns(columns) {
        this.columns = columns;
        S8WebFront.setS8Elements(this.wrapperNode, columns);
    }


    /** render */
    S8_render() { }


    /** dispose */
    S8_dispose() { }
}

File: sources/com/s8/fwks/palm/components/boxes/b2/B2ColumnsStruct.java
package com.s8.fwks.palm.components.boxes.b2;

import com.s8.api.web.S8WebFront;
import com.s8.fwks.palm.components.PalmWebSources;
import com.s8.pkgs.ui.carbide.HTML_S8WebNode;


/**
 * 
 * @author pierreconvert
 *
 */
public class B2ColumnsStruct extends HTML_S8WebNode {
	
	
	
	public final static String WEBPATH = PalmWebSources.ROOT_WEBPATH + "/boxes/b2/B2ColumnsStruct";
	
	

	
	/**
	 * 
	 * @param front
	 */
	
	public B2ColumnsStruct(S8WebFront front) {
		super(front, WEBPATH);
	}
	
	


	/**
	 * Columns
	 * @param size
	 */
	public void setColumns(B2Column... columns) {
		vertex.outbound().setObjectListField("columns", columns);
	}
	
	
}
